src/Microsoft.OData.Core/Uri/ODataUriConversionUtils.cs

```diff
@@ -577,7 +579,18 @@ private static void WriteJsonLightLiteral(IEdmModel model, ODataMessageWriterSet
                 new ODataJsonLightOutputContext(textWriter, messageInfo, messageWriterSettings))
             {
                 ODataJsonLightValueSerializer jsonLightValueSerializer = new ODataJsonLightValueSerializer(jsonOutputContext);
-                writeValue(jsonLightValueSerializer);
+
+                if (ignoreDuplicatePropertyNameChecker)

```

Comment:
>ignoreDuplicatePropertyNameChecker [](http://example.com/codeflow?start=20&length=34)

why don't use the setting in ODataMessageWriterSettings and avoid adding this parameter?

Reply:
I have changed the naming. This boolean was supposed to distinguish between `Resource` and `Collection` values. I have renamed to `isResourceValue`